cpp/src/arrow/json/chunked_builder.cc

```diff
@@ -466,5 +484,13 @@ Status MakeChunkedArrayBuilder(const std::shared_ptr<TaskGroup>& task_group,
   return Status::OK();
 }
 
+Status MakeChunkedArrayBuilder(const std::shared_ptr<TaskGroup>& task_group,
```

Comment:
IIUC, this overload is only called for top-level fields of an explicit schema? Add a comment explaining this?